spyder/plugins/explorer/widgets/fileassociations.py

```diff
@@ -0,0 +1,525 @@
+    """Dialog for selection of installed system/user applications."""
+
+    def __init__(self, parent=None):
+        """Dialog for selection of installed system/user applications."""
+        super(ApplicationsDialog, self).__init__(parent=parent)
+
+        # Widgets
+        self.label = QLabel()
+        self.label_browse = QLabel()
+        self.edit_filter = QLineEdit()
+        self.list = QListWidget()
+        self.button_browse = QPushButton(_('Browse...'))
+        self.button_box = QDialogButtonBox(QDialogButtonBox.Ok
+                                           | QDialogButtonBox.Cancel)
+        self.button_ok = self.button_box.button(QDialogButtonBox.Ok)
+        self.button_cancel = self.button_box.button(QDialogButtonBox.Cancel)
+
+        # Widget setup
+        self.setWindowTitle(_('Applications dialog'))
+        self.edit_filter.setPlaceholderText(_('Type to filter by name...'))
+        self.list.setIconSize(QSize(16, 16))  # FIXME: Use metrics
+
+        # Layout
+        layout = QVBoxLayout()
+        layout.addWidget(self.label)
+        layout.addWidget(self.edit_filter)
+        layout.addWidget(self.list)
+        layout_browse = QHBoxLayout()
+        layout_browse.addWidget(self.button_browse)
+        layout_browse.addWidget(self.label_browse)
+        layout.addLayout(layout_browse)
+        layout.addSpacing(12)  # FIXME: Use metrics
+        layout.addWidget(self.button_box)
+        self.setLayout(layout)
+
+        # Signals
+        self.edit_filter.textChanged.connect(self.filter)
+        self.button_browse.clicked.connect(lambda x: self.browse())
+        self.button_ok.clicked.connect(self.accept)
+        self.button_cancel.clicked.connect(self.reject)
+
+        self.setup()
+
+    def setup(self, applications=None):
+        """Load installed applications."""
+        QApplication.setOverrideCursor(QCursor(Qt.WaitCursor))
+        self.list.clear()
+        if applications is None:
+            apps = get_installed_applications()
+        else:
+            apps = applications
+
+        for app in sorted(apps, key=lambda x: x.lower()):
+            fpath = apps[app]
+            icon = get_application_icon(fpath)
+            item = QListWidgetItem(icon, app)
+            item.setToolTip(fpath)
+            item.fpath = fpath
+            self.list.addItem(item)
+
+        # FIXME: Use metrics
+        self.list.setMinimumWidth(self.list.sizeHintForColumn(0) + 24)
+        QApplication.restoreOverrideCursor()
+
+    def browse(self, fpath=None):
+        """Prompt user to select an application not found on the list."""
+        app = None
+        item = None
+
+        if sys.platform == 'darwin':
+            if fpath is None:
+                basedir = '/Applications/'
+                filters = _('Applications (*.app)')
+                title = _('Select application')
+                fpath, __ = getopenfilename(self, title, basedir, filters)
+
+            if fpath and fpath.endswith('.app'):
```

Comment:
Should we test that `fpath` is also a directory here?

Reply:
Will do